### ISSUE TITLE
NVDA fails to launch in Windows PE environment with no error output

### ISSUE THREAD

**Issue by akash07k** [nvaccess/nvda#19537]
### Brief summary

NVDA does not start when executed in a Windows PE (WinPE) environment. Running `nvda.exe` from the command line produces no output, no error message, and no visible process, making it impossible to diagnose missing dependencies or incompatibilities.
Edit: %temp%\nvda.log is created which is attached and pasted in log section below

### Prerequisites

1. Create a bootable USB or ISO using:
   * Windows installation media or
   * A custom Windows PE image (VM recommended)
2. Copy either:

   * NVDA installer or
   * NVDA portable package
     to the USB stick (Writable partition)

### Steps

1. Boot into Windows PE (for example, from a Windows installation ISO).
2. When the language selection screen appears, optionally start Narrator using **Ctrl + Windows + Enter**.
3. Press **Shift + F10** to open Command Prompt.
4. Run NVDA manually, e.g.:

   ```
   X:\NVDA\nvda.exe
   ```
5. Observe the result.

### Actual behavior

* NVDA does not launch.
* No error message is displayed.
* No output is written to the console.
* No visible NVDA process appears to start.
* `%temp%\nvda.log` is created.

### Expected behavior

One of the following should occur:

1. Preferred: NVDA launches successfully in the WinPE environment.
2. Alternative: NVDA fails gracefully and reports:

   * Missing DLLs
   * Unsupported APIs
   * Required Windows components not present in WinPE

This information would allow users or developers to manually integrate the required dependencies into the WinPE `.wim` image.

### NVDA logs, crash dumps and other attachments

```
INFO - __main__ (00:50:36.107) - MainThread (2244):
Starting NVDA version source-master-93f8c66 AMD64
INFO - core.main (00:50:36.126) - MainThread (2244):
Config dir: C:\n\userConfig
INFO - config.ConfigManager._loadConfig (00:50:36.129) - MainThread (2244):
Loading config: C:\n\userConfig\nvda.ini
CRITICAL - __main__ (00:50:36.557) - MainThread (2244):
core failure
Traceback (most recent call last):
  File "ctypes\__init__.pyc", line 470, in __getattr__
  File "ctypes\__init__.pyc", line 361, in __init__
  File "ctypes\__init__.pyc", line 379, in _load_library
FileNotFoundError: Could not find module 'bthprops.cpl' (or one of its dependencies). Try using the full path with constructor syntax.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "nvda.pyw", line 309, in <module>
  File "core.pyc", line 717, in main
  File "NVDAHelper\__init__.pyc", line 49, in <module>
  File "eventHandler.pyc", line 13, in <module>
  File "api.pyc", line 15, in <module>
  File "review.pyc", line 15, in <module>
  File "NVDAObjects\__init__.pyc", line 26, in <module>
  File "displayModel.pyc", line 16, in <module>
  File "mouseHandler.pyc", line 9, in <module>
  File "gui\__init__.pyc", line 17, in <module>
  File "ui.pyc", line 30, in <module>
  File "braille.pyc", line 67, in <module>
  File "hwPortUtils.pyc", line 22, in <module>
  File "winBindings\bthprops.pyc", line 19, in <module>
  File "ctypes\__init__.pyc", line 477, in __getitem__
  File "ctypes\__init__.pyc", line 472, in __getattr__
AttributeError: bthprops.cpl
```

[nvda.log](https://github.com/user-attachments/files/24985404/nvda.log)

### NVDA type

portable copy, installer/launcher copy

### NVDA version

nvda 64 bit snapshot_source-master-e5d0196

### Have you tried any other versions of NVDA? If so, please report their behaviors.

Happens in every 64 build. (32 bit NVDA is not compattible since Windows PE is now 64 bit only and lacks 32 bit components)

### Windows version

Windows PE (booted via Windows installation ISO) (ISO generated from Windows 11 25H2 26200.7701)

### Name and version of other software in use when reproducing the issue

* Platform: Windows PE (booted via Windows installation ISO)
* Execution method: Command Prompt in WinPE
* Architecture: Same as WinPE image (x64)

### Additional Notes

* Windows Narrator runs successfully in the same WinPE session.
* The silent failure makes troubleshooting difficult, as there is no diagnostic feedback.
* Even a basic stderr or message box output would significantly improve debuggability.

### Does the issue still occur after restarting your computer?

I have restarted my computer and the issue still occurs

### If NVDA add-ons are disabled, is your problem still occurring?

I do not have any NVDA add-ons installed

### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

I have not run the COM Registration Fixing Tool and it can't even be run since NVDA can't run.

**Comment by SaschaCowley** [comment]
Hi @akash07k,
> 32 bit NVDA is not compattible since Windows PE is now 64 bit only and lacks 32 bit components
This means that the launcher will not work, as it is currently an x86 executable.
After attempting to run an x64 portable, is `%temp%\nvda.log` created?

**Comment by akash07k** [comment]
Hi @SaschaCowley,
Yes, I just checked, `%temp%\nvda.log` is created.
below is the log content:
Seems that it tries to find for `bthprops.cpl` which is not available in Windows PE.

```
INFO - __main__ (00:50:36.107) - MainThread (2244):
Starting NVDA version source-master-93f8c66 AMD64
INFO - core.main (00:50:36.126) - MainThread (2244):
Config dir: C:\n\userConfig
INFO - config.ConfigManager._loadConfig (00:50:36.129) - MainThread (2244):
Loading config: C:\n\userConfig\nvda.ini
CRITICAL - __main__ (00:50:36.557) - MainThread (2244):
core failure
Traceback (most recent call last):
  File "ctypes\__init__.pyc", line 470, in __getattr__
  File "ctypes\__init__.pyc", line 361, in __init__
  File "ctypes\__init__.pyc", line 379, in _load_library
FileNotFoundError: Could not find module 'bthprops.cpl' (or one of its dependencies). Try using the full path with constructor syntax.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "nvda.pyw", line 309, in <module>
  File "core.pyc", line 717, in main
  File "NVDAHelper\__init__.pyc", line 49, in <module>
  File "eventHandler.pyc", line 13, in <module>
  File "api.pyc", line 15, in <module>
  File "review.pyc", line 15, in <module>
  File "NVDAObjects\__init__.pyc", line 26, in <module>
  File "displayModel.pyc", line 16, in <module>
  File "mouseHandler.pyc", line 9, in <module>
  File "gui\__init__.pyc", line 17, in <module>
  File "ui.pyc", line 30, in <module>
  File "braille.pyc", line 67, in <module>
  File "hwPortUtils.pyc", line 22, in <module>
  File "winBindings\bthprops.pyc", line 19, in <module>
  File "ctypes\__init__.pyc", line 477, in __getitem__
  File "ctypes\__init__.pyc", line 472, in __getattr__
AttributeError: bthprops.cpl

```

[nvda.log](https://github.com/user-attachments/files/24985398/nvda.log)

> Hi [@akash07k](https://github.com/akash07k),
> 
> > 32 bit NVDA is not compattible since Windows PE is now 64 bit only and lacks 32 bit components
> > This means that the launcher will not work, as it is currently an x86 executable.
> > After attempting to run an x64 portable, is `%temp%\nvda.log` created?

**Comment by seanbudd** [comment]
To handle this we should start with more graceful failing, reporting what modules are missing.
It's worth investigating if we can improve import handing in general and create a minimum environment for PE through checking what features are available.

**Comment by akash07k** [comment]
Hi @seanbudd ,

That's right.
I'll try to remove the dependency from `bthprops.cpl` in my source copy once and will see what else is missing, but I'm wondering that it is going to be a rabbit hole for me 😊
Nevertheless, the idea of creating a minimum environment for PE through checking what features are available seems a viable one.
Please keep us updated if NVAccess plans to put some time in this regard.
> To handle this we should start with more graceful failing, reporting what modules are missing. It's worth investigating if we can improve import handing in general and create a minimum environment for PE through checking what features are available.

**Comment by akash07k** [comment]
@SaschaCowley , @seanbudd ,

I've finally opened a PR for this after months of investigation, implementation and efforts.
Hope NVAccess accepts it.
[Allow NVDA to start when optional Windows components are unavailable- #20842](https://github.com/nvaccess/nvda/pull/20842)
